src/MongoDB.Driver.Core/Core/Bindings/IChannel.cs

```diff
@@ -466,7 +466,7 @@ public interface IChannel : IDisposable
             MessageEncoderSettings messageEncoderSettings,
             WriteConcern writeConcern,
             BsonDocument query,
-            BsonDocument update,
+            BsonValue update,
```

Comment:
This should remain BsonDocument.

IChannel.Update corresponds to an OP_UPDATE message, and the OP_UPDATE message is not being changed to support pipeline updates.

This also means several other files require no changes. I'll note them.